feat(idempotency): handle lambda timeout scenarios for INPROGRESS records (#1387)
awslabs/aws-lambda-powertools-python: docs/utilities/idempotency.md

* Ensure Lambda handler returns the same result when called with the same payload
* Select a subset of the event as the idempotency key using JMESPath expressions
* Set a time window in which records with the same payload should be considered duplicates

> * Expires in-progress executions when the Lambda handler times out

Reviewer: ```suggestion
* Expires in-progress executions if the Lambda function times out halfway through
```
